src/components/stepforms/IfThenElseStepForm.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <div class="ifthenelse-form">
+    <StepFormHeader :title="title" :stepName="this.editedStep.name" />
+    <InputTextWidget
+      class="newColumnInput"
+      v-model="editedStep.newColumn"
+      name="New colum:"
+      placeholder="Enter a name"
+      data-path=".newColumn"
+      :errors="errors"
+      :warning="duplicateColumnName"
+    />
+    <IfThenElseWidget :value="ifthenelse" @input="updateIfThenElse" :errors="errors" />
+    <StepFormButtonbar />
+  </div>
+</template>
+
+<script lang="ts">
+import _ from 'lodash';
+import { Prop } from 'vue-property-decorator';
+
+import { StepFormComponent } from '@/components/formlib';
+import { castFilterStepTreeValue } from '@/components/stepforms/convert-filter-step-tree.ts';
+import IfThenElseWidget from '@/components/stepforms/widgets/IfThenElseWidget.vue';
+import InputTextWidget from '@/components/stepforms/widgets/InputText.vue';
+import { ColumnTypeMapping } from '@/lib/dataset';
+import { IfThenElseStep } from '@/lib/steps';
+import { VQBModule } from '@/store';
+
+import BaseStepForm from './StepForm.vue';
+
+@StepFormComponent({
+  vqbstep: 'ifthenelse',
+  name: 'ifthenelse-step-form',
+  components: {
+    IfThenElseWidget,
+    InputTextWidget,
+  },
+})
+export default class IfThenElseStepForm extends BaseStepForm<IfThenElseStep> {
+  @Prop({
+    type: Object,
+    default: () => ({
+      name: 'ifthenelse',
+      newColumn: '',
+      if: { column: '', value: '', operator: 'eq' },
+      then: '',
+      else: '',
+    }),
+  })
+  initialStepValue!: IfThenElseStep;
+
+  @VQBModule.Getter columnTypes!: ColumnTypeMapping;
+
+  readonly title: string = 'Add a conditional column';
+
+  ifthenelse = _.omit({ ...this.initialStepValue }, ['name', 'newColumn']);
```

Comment:
To avoid importing all lodash object `_`, please import only `omit` function from lodash with `import _omit from 'lodash/omit'`

Reply:
Even better: avoid using lodash at all, with destructuring:
`const {name , newColumn , ...ifthenelse} = this.initialStepValue`

Reply:
I couldn't use the destructuring syntax inside the component. So I'm implementing your first suggestion